addons/portal/models/portal_mixin.py

```diff
@@ -14,7 +15,8 @@ class PortalMixin(models.AbstractModel):
     access_url = fields.Char(
         'Portal Access URL', compute='_compute_access_url',
         help='Customer Portal URL')
-    access_token = fields.Char('Security Token', copy=False)
+    access_token = fields.Char('Security Token (readonly)', copy=False)
```

Comment:
try to only use one token, which will simplify the logic in many places: no need for dual _check_token, etc.